test/e2e/shoot/internal/logging/event_logging.go

```diff
@@ -0,0 +1,264 @@
+	v.seedEvent = getEventFor("seed", v.shootNamespace)
+	v.shootEvent = getEventFor("shoot", kubeSystemNamespace)
+
+	Eventually(func(g Gomega) {
+		g.Expect(v.ShootFramework.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: gardenNamesapce, Name: fluentBitDaemonSetName}, &appsv1.DaemonSet{})).To(Succeed())
+		g.Expect(v.ShootFramework.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: v.shootNamespace, Name: lokiName}, &appsv1.StatefulSet{})).To(Succeed())
+		g.Expect(v.ShootFramework.SeedClient.Client().Get(ctx, client.ObjectKey{Namespace: v.shootNamespace, Name: eventLogger}, &appsv1.Deployment{})).To(Succeed())
+	}).Should(Succeed())
+
+	By("Waite for the shoot Loki and Event-Logger to become ready")
+	Expect(v.WaitUntilStatefulSetIsRunning(ctx, lokiName, v.shootNamespace, v.ShootFramework.SeedClient)).To(Succeed())
+	Expect(v.WaitUntilDeploymentIsReady(ctx, eventLogger, v.shootNamespace, v.ShootFramework.SeedClient)).To(Succeed())
+}
+
+// Prepare is called after Before and it makes all preparation for the test.
+func (v *EventLoggingVerifier) Prepare(ctx context.Context) {
+	By("Create seed and shoot events")
+	Eventually(func(g Gomega) {
+		g.Expect(v.ShootFramework.SeedClient.Client().Create(ctx, v.seedEvent)).To(Succeed())
+		g.Expect(v.ShootFramework.ShootClient.Client().Create(ctx, v.shootEvent)).To(Succeed())
+	}).Should(Succeed())
+}
+
+// Expect is the the process were we expect to get the correct result from the logging test.
+func (v *EventLoggingVerifier) Expect(ctx context.Context) {
+	By("Wait until Loki receive seed event")
+	Expect(waitUntilLokiReceivesEvent(ctx, v.Logger, 30*time.Second, lokiLabels, operatorID, v.shootNamespace, `origin_extracted="seed",source="event-logger",sourceHost="seed"`, []string{eventMessageForSeed}, time.Now().Add(-5*time.Minute), v.ShootFramework.SeedClient)).To(Succeed())
+
+	By("Wait until Loki receive shoot event")
+	Expect(waitUntilLokiReceivesEvent(ctx, v.Logger, 30*time.Second, lokiLabels, operatorID, v.shootNamespace, `origin_extracted="shoot",source="event-logger",sourceHost="shoot"`, []string{eventMessageForShoot}, time.Now().Add(-5*time.Minute), v.ShootFramework.SeedClient)).To(Succeed())
+}
+
+// After cleans all resources created in Prepare function
+func (v *EventLoggingVerifier) After(ctx context.Context) {
+	By("Delete seed and shoot events")
+	Eventually(func(g Gomega) {
+		g.Expect(v.ShootFramework.SeedClient.Client().Delete(ctx, v.seedEvent)).To(Succeed())
+		g.Expect(v.ShootFramework.ShootClient.Client().Delete(ctx, v.shootEvent)).To(Succeed())
+	}).Should(Succeed())
+}
+
+func getEventFor(clusterType, namespace string) *corev1.Event {
+	clusterType = strings.ToLower(clusterType)
+
+	event := &corev1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+		},
+		InvolvedObject: corev1.ObjectReference{
+			Kind:      clusterType,
+			Namespace: namespace,
+			Name:      clusterType + "-" + eventLoggerName + "-testing",
+		},
```

Comment:
This doesn't make any sense. There is no such object in the system.
Either drop `InvolvedObject` entirely or reference an existing object.

Reply:
I see that the InvolvedObject is not optional so I guess I cannot omit it.
And I am not searching for logical sense, I need this resource only for testing purposes. Could you tell me which object in the shoot should I address when creating the event?